### DISCUSSION TITLE
Stop github actions running on a fork

### DISCUSSION THREAD

**Discussion by jimbob88** [community/community#26704] [Actions]
<p>Hey everyone,</p>
<p>I have a fork of <a href="https://github.com/danielkrupinski/Osiris" title="Master Osiris" rel="nofollow noopener">danielkrupinski\Osiris</a> on my Github, <a href="https://github.com/jimbob88/Osiris" title="My Fork of Osiris" rel="nofollow noopener">here</a> and recently Daniel pushed <a href="https://github.com/danielkrupinski/Osiris/pull/1302" title="roboot git pull request" rel="nofollow noopener">a pull request</a> which added a new action for testing the building of Osiris. Personally, I don’t want to have these actions running on my Fork, is there an easy way to freeze actions so as to stop my fork using up my Action time… By the way, I expect the main.yml to updated quite frequently on his master branch soon as the pull request mentions adding things such as artifacting + automatic release uploading! If I just delete main.yml the pull request, which is currrently automatic, will always conflict!</p>
<p>Also, is it possible to have my own github action, instead of his, because with <a href="https://github.com/jimbob88/Osiris/releases" title="My Releases Section" rel="nofollow noopener">my releases</a>, I build for AVX, AVX2, IA32 &amp; SSE2, and I have a <a href="https://pastebin.com/aD8nrcUs" title="My Custom Actions" rel="nofollow noopener">main.yml</a> and obviously this would conflict with my automatic pull system, as it did when I first had these github actions and having this main.yml would make it so I would have to go through the round-about route of allowing his main.yml to update (i.e. delete my main.yml, pushing his pull request, adding my custom main.yml back), feel free to point me in the right direction, if I am wrong here though!</p>
<p>Thanks in advance,</p>
<p>Jim</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/jimbob88">@jimbob88</a> ,</p>
<p>If you do not want any workflows run on your repository, you can navigate to <strong>Settings</strong> &gt; <strong>Actions</strong> &gt; <strong>Actions permissions</strong> to <strong>Disable Actions for this repository</strong>.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/9/946ab7e1f7c46ac155aaf47a7a479a67471af157.png" data-download-href="/uploads/short-url/laXj7V3usn3AosFM7F3KSTDefen.png?dl=1" title="disable_actions.PNG" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/46307996/181099312-21af1e9f-47ad-4ba2-9fac-7da93aa21882.png" alt="disable_actions.PNG" data-base62-sha1="laXj7V3usn3AosFM7F3KSTDefen" width="690" height="293" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/9/946ab7e1f7c46ac155aaf47a7a479a67471af157_2_690x293.png, https://aws1.discourse-cdn.com/github/original/2X/9/946ab7e1f7c46ac155aaf47a7a479a67471af157.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/9/946ab7e1f7c46ac155aaf47a7a479a67471af157.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/9/946ab7e1f7c46ac155aaf47a7a479a67471af157_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">disable_actions.PNG</span><span class="informations">1001×426 27.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>However, as you mentioned, you may need to run some of yourself workflows, so this may be not a perfect way. And we have no methods to allow us to tell the PRs only merge the files we want and do not merge the files we do not want.</p>
<p>As a workaround, maybe you can try this:</p>
<ol>
<li>
<p>Keep all the workflow YAML files in your forked repository, include yours and other people’s.</p>
</li>
<li>
<p>In your workflows, add the following <a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idif" rel="nofollow noopener"><strong>if conditional</strong></a> for each job.</p>
<p>if: github.repository == ‘jimbob88/Osiris’</p>
</li>
<li>
<p>In other workflows, add the following if conditional for each job.</p>
<p>if: github.repository != ‘jimbob88/Osiris’</p>
</li>
</ol>
<p>The property <strong>github.repository</strong> of the <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context" rel="nofollow noopener"><strong>github context</strong></a> can return name of the repository where the workflow is running. Of course, you also need to merge and sync the same configurations to the original repository.</p>
<p>This is the only way I can think of, hope it can help you.</p>

**Comment by keeganwitt** [comment]
<p>This worked for my use case, where I only forked a repo to make a PR against the upstream repo.  I have no interest in running the scheduled junk they run upstream.</p>
<p>But it seems like having it disabled really ought to be the default.  It’s that way for other systems, like Travis.  I know the permissions model is different there, but still, it shows folks are fine with having to check a box before their CI config runs.</p>

**Comment by rhaschke** [comment]
<p>I fully agree. You should actively opt-in for github actions to run, particularly on forks! Currently I’m struggling with a GHA workflow that is scheduled for a nightly run … and failing, causing many spurious email notifications for a repo that I just forked.<br>
Disabling actions completely isn’t an option for me, because I do want to validate my stuff <em>before</em> filing a PR against the main repo.</p>

**Comment by jedwards1211** [comment]
<p>I landed here because I made a PR to babel and I got an email about an action failing.  I thought my PR had failed, but when I went to it I didn’t see any CI failures…I was confused for a moment before I realized that the github actions were running separately in my fork, and one was failing to publish to their Verdaccio registry because obviously I don’t have permissions.</p>
<p>My reaction was:</p>
<ul>
<li>Whoa, this is a big surprise compared to how all other CI platforms work</li>
<li>This probably isn’t what the maintainers wanted to happen either</li>
<li>Did this count against a quota on my account, accidentally?</li>
<li>This feels like a security exploit waiting to happen</li>
</ul>
<p>Should absolutely be disabled by default.</p>

**Comment by josecelano** [comment]
<p>It seems you can <a href="https://docs.github.com/en/actions/managing-workflow-runs/disabling-and-enabling-a-workflow" rel="noopener nofollow ugc">disable a workflow manually</a>.</p>

**Comment by pombredanne** [comment]
FWIW I have 5K+ repos. I cannot go through each of them one by one.... I now found out that there are random jobs running on some of my forks... https://github.com/actions/runner/issues/2063

**Comment by krs0** [comment]
Default should be not to run these things. I only found out months later that Scheduled actions were running daily for ~1h on a repo I forked. Complete waste of resources.

**Comment by rickstaa** [comment]
Is there a way to create a feature request for changing the default behaviour to opt-in instead of opt-out 🤔? Many people use @BrightRan's workaround (see https://github.com/search?q=if%3A+github.repository+%3D%3D+%27&type=code), but in most cases, I encounter, only a small group of people (core maintainers) want to enable actions on their forks. In contrast, the majority (i.e. the primary users) want them to be disabled. Because of this, I think the opt-in behaviour suggested by others above makes more sense.

**Comment by pombredanne** [comment]
There is another issue: there is a cost attached to running these non-desired jobs : https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions

Could it be that GitHub's interest is to have user use and eventually pay for more of these?

**Comment by rickstaa** [comment]
I just noticed that GitHub already implemented this feature, but it is not clearly documented yet. Currently, when you fork a repository containing actions, you first have to opt-in for these actions to run. You will get a prompt for this the first time you open the action tab.

![image](https://user-images.githubusercontent.com/17570430/224348663-7a50746f-73ae-44c2-b61f-98d8522a1690.png)

**Reply by laurent22** [reply]
The problem is that people don't really know what they are going to enable or disable, so as far as I can see many click that big green button anyway, because "why not". So this is a good solution for people who fork and really don't want to run any action, but a bad solution for repo owner who don't want any fork to trigger their workflows.

**Reply by rickstaa** [reply]
> The problem is that people don't really know what they are going to enable or disable, so as far as I can see many click that big green button anyway, because "why not". So this is a good solution for people who fork and really don't want to run any action, but a bad solution for repo owner who don't want any fork to trigger their workflows.

If I recall it correctly, the prompt only shows up when users click on the actions tab after forking, not during the forking pipeline 🤔. You can test this behaviour if you like. I, however, agree with you that the workflow can be improved. I think having an option to only enable certain actions instead of enabling them all would be an improvement.

**Reply by laurent22** [reply]
Not sure exactly how it works, but at the moment we're being spammed by people forking the repo, which in turns trigger the CLA workflow, and that sends notifications to all the previous contributors.

I've tried changing the workflows so that they run only from our repo [with this change](https://github.com/laurent22/joplin/commit/a481bf1b537e6b9294f7c1a4d5ed95f9675523e2) but that didn't help, and we're still being spammed.

I feel the solution would be some switch to say we only accept workflows that originate from specific repositories. I don't see the point of our workflows running for other people's forks.

**Reply by rickstaa** [reply]
> Not sure exactly how it works, but at the moment we're being spammed by people forking the repo, which in turns trigger the CLA workflow, and that sends notifications to all the previous contributors.
> 
> I've tried changing the workflows so that they run only from our repo [with this change](https://github.com/laurent22/joplin/commit/a481bf1b537e6b9294f7c1a4d5ed95f9675523e2) but that didn't help, and we're still being spammed.
> 
> I feel the solution would be some switch to say we only accept workflows that originate from specific repositories. I don't see the point of our workflows running for other people's forks.

I quickly tested it out. It seems that the behaviour I described is correct:

![image](https://user-images.githubusercontent.com/17570430/229197767-f276d16c-456d-4b76-821b-eba5108d6c3a.png)

Meaning that the spam you receive is only from the people who forked the repository before GitHub implemented the opt-in behaviour. The condition you use in your action should disable the fork runs for these people. If that doesn't work, you can maybe try using environmental variables (see https://github.com/sdr-enthusiasts/plane-alert-db/blob/321424922c3b00959d99f30e53ec36ad45609e62/.github/workflows/create_db_derivatives.yaml#L18) 🤔.

**Reply by maphew** [reply]
> I just noticed that GitHub already implemented this feature, but it is not clearly documented yet. Currently, when you fork a repository containing actions, you first have to opt-in for these actions to run. You will get a prompt for this the first time you open the action tab.

I did not get this prompt. Did something change in the year since this post?

**Reply by tredondo** [reply]
March 2026 - I did NOT see this prompt when I forked a large repo that has GitHub Actions.

**Reply by maifeeulasad** [reply]
Update june 2026, this awesome feature has been disabled!! Now every action runs by default on forks. Which is happening for about a year now.

**Comment by Vyvy-vi** [comment]
add a check in the workflow that tests the repository name

**Comment by jonian** [comment]
Still no way to disable upstream actions and allow only the actions in your fork to run. Adding checks does not make sense, you can just delete those actions.

Why not provide an option where you select only the actions that you want to run in your repo.

**Comment by christoferd** [comment]
How can I disable workflows automatically running in Github?

This has not been answered for the latest version of the UI :-(

![image](https://github.com/user-attachments/assets/2f8f8e62-b0ee-440d-9334-72c2f7ab6587)

**Reply by Mephi78** [reply]
Actions > General

![{441F252B-1355-4C61-8EA5-6175BDEAEA0C}](https://github.com/user-attachments/assets/fc2c99e9-30f0-4f7a-9d91-47cb475dea8d)

Although GitHub states, workflows should be [disabled by default on forked public repos](https://docs.github.com/en/actions/managing-workflow-runs-and-deployments/managing-workflow-runs/disabling-and-enabling-a-workflow), from recent experience it might happen, that they in fact are not or are re-enabled after hitting sync fork.

**Reply by mootari** [reply]
I can confirm that workflows ran without me enabling them explicitly, and I do remember clicking "Sync fork" shortly after forking.

**Reply by egrosdou01** [reply]
You still receive notifications even if the "Disable actions" is checked.

**Comment by btzdnl** [comment]
Disabling actions altogether didn’t work for me. For a scheduled job ([this one](https://github.com/discourse/discourse/blob/main/.github/workflows/stale-pr-closer.yml), but in my now-deleted fork) I still received notifications, now telling me “Close stale PRs: All jobs were cancelled”.

**Reply by shaju007** [reply]
Another option is to add ```if: github.repository == repo-name``` on the workflow

**Comment by clicktodev** [comment]
i think github now disables actions for new forks by default

**Reply by oliviertassinari** [reply]
Right, it seems to be the case. What I get after a clean fork:

<img width="1026" height="480" alt="SCR-20251116-jeyk" src="https://github.com/user-attachments/assets/589fdf65-a4f7-47a6-a153-c526ea0281a5" />

**Reply by clicktodev** [reply]
yeah, you have to disable it manually for existing forks

**Comment by maifeeulasad** [comment]
Update june 2026, this awesome feature has been disabled!! Now every action runs by default on forks. Which is happening for about a year now.

Sad unga bunga noise

<img width="640" height="454" alt="image" src="https://github.com/user-attachments/assets/32ef4d52-2de0-4c6d-a10e-d685ec70536e" />

**Reply by dimateos** [reply]
~~I just cloned `vscode` and all the workflows were active...~~
~~The disclaimer window only seems to appear on repos with no actions yet? So once enabled forks skip it? idk~~

~~unga bunga indeed~~

Nah it was an older fork I forgot, I just tried and fork two new repos with actions enabled and the disclaimer page in actions was there.
